### ISSUE TITLE
`indent-binary-ops`: Expects 0 intendation on multiline re-assignment

### ISSUE THREAD

**Issue by anne-gropler** [eslint-stylistic/eslint-stylistic#546]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

I noticed that the indentation of the multiline assignment depends on whether it is an assignment (line starts with `const` or `let`) or an re-assignment (line starts just with the variable itself).

Expected:

```ts
const a = 1
  + 2;

let b = 1
  + 2;

b = 2
  + 2; // expected
```

Actual:

```ts
const a = 1
  + 2;

let b = 1
  + 2;

b = 2
+ 2; // actual
```

### Reproduction

https://stackblitz.com/edit/github-fmnurc?file=src%2Findex.ts,README.md,eslint.config.js

### Contributes

- ~~[ ] I am willing to submit a PR to fix this issue~~
- ~~[ ] I am willing to submit a PR with failing tests~~
